src/plugins/discover/public/components/discover_grid/render_custom_toolbar.scss

```diff
@@ -33,7 +32,7 @@
 }
 
 .dscGridToolbarControlGroup {
-  border: $euiBorderThin;
+  box-shadow: inset 0 0 0 $euiBorderWidthThin $euiBorderColor;
```

Comment:
Using `border` here makes the buttons 34px instead of 32px, so they're slightly larger than the ones next to them. Using `box-shadow` instead seems to avoid this.